File: pset6/vigenere/vigenere.py
import sys
from cs50 import get_string


if len(sys.argv) != 2 or sys.argv[1].isalpha() == False:
    print("Usage: ./vigenere k")
    exit(1)

def keyToNum(keychar):
    """Convert keyword into key"""
    if keychar.isupper():
        return ord(keychar)-65
    elif keychar.islower():
        return ord(keychar)-97


keyword = sys.argv[1]
plain = get_string("plaintext: ")
cipher = ""
counter = 0
for i in range(len(plain)):
    if plain[i].isupper():
        index = ord(plain[i])-65
        temp = (index + keyToNum(keyword[counter % len(keyword)]))% 26
        temp += 65
        cipher += chr(temp)
        counter +=1

    elif plain[i].islower():
        index = ord(plain[i])-97
        temp = (index + keyToNum(keyword[counter % len(keyword)]))% 26
        temp += 97
        cipher += chr(temp)
        counter +=1
    else:
        cipher += plain[i]
print("ciphertext: " + cipher)

File: pset2/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int keyToNum(char keyword);
int checkAlpha(string key);

int main(int argc, string argv[])
{
    if(argc != 2 || !checkAlpha(argv[1]))
    {
        printf("Usage: ./vigenere k\n");
        return 1;
    }
    else
    {
        string s = get_string();
        int n = strlen(argv[1]);
        int sn = strlen(s);
        char cipher[sn+1];
        for(int i = 0 ; i < sn ; i++)
        {
            char ori = s[i];
            if(isupper(ori))
            {
                ori -= 65;
                int ans = (ori + keyToNum(argv[1][i%n])) % 26 ;
                ans += 65;
                cipher[i] = (char)ans;
            }
            else if(islower(ori))
            {
                ori -= 97;
                int ans = (ori + keyToNum(argv[1][i%n])) % 26 ;
                ans += 97;
                cipher[i] = (char)ans;
            }
            else
            {
                cipher[i] = s[i];
            }
        }
    printf("plaintext: %s\n",s);
    printf("ciphertext: %s\n",cipher);
    return 0;
    }
}

int keyToNum(char keyword)
{
    int outp = 0;
    if(isalpha(keyword))
    {
        outp = toupper(keyword) - 65;
    }
    else
    {
        outp = keyword;
    }
    return outp;
}

int checkAlpha(string key)
{
    for(int i = 0, n = strlen(key); i < n; i++)
    {
        if(isalpha(key[i]) == false)
        {
            return false;
        }
    }
    return true;
}